File: 文件夹-遍历/walk.py
import os
import sys

def walk(dir):
    for root, dirs, files in os.walk(dir):
        for file in files:
            print(os.path.join(root, file))
        for dir in dirs:
            print(os.path.join(root, dir))


currentDir = os.path.split(os.path.realpath(sys.argv[0]))[0]
walk(currentDir);

File: 文件夹-遍历/walk.js
const path = require("path");
const fs = require("fs");

function walk(root) {
    const dirs = fs.readdirSync(root);
    dirs.forEach(dir => {
        const fullPath = path.join(root, dir);
        const sta = fs.statSync(fullPath);
        if(sta.isFile()){
            return console.log(fullPath);
        }
        console.log(fullPath);
        walk(fullPath);
    });
}

walk(__dirname);

